blockindex/liquidstaking_bucket.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2023 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package blockindex
+
+import (
+	"math/big"
+	"time"
+
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/iotexproject/iotex-core/blockindex/indexpb"
+	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type (
+	// BucketInfo is the bucket information
+	BucketInfo struct {
+		TypeIndex  uint64
+		CreatedAt  time.Time
+		UnlockedAt *time.Time
+		UnstakedAt *time.Time
+		Delegate   string
+		Owner      string
+	}
+
+	// BucketType is the bucket type
+	BucketType struct {
+		Amount      *big.Int
+		Duration    time.Duration
+		ActivatedAt *time.Time
+	}
+
+	// Bucket is the bucket information including bucket type and bucket info
+	Bucket struct {
```

Comment:
this is same as `staking.VoteBucket()` except `Candidate` field? should use the existing struct if possible

Reply:
The `Candidate` field in `staking.VoteBucket` is `Address`, and this one is a `string` representing candidate name. It may not be possible to convert name to address in the liquid staking indexer, due to lack of candidate register info. Implementing the convert in staking protocol may be more reasonable